cognite_toolkit/_cdf_tk/resource_classes/transformations.py

```diff
@@ -0,0 +1,151 @@
+    audience: str | None = Field(default=None, description="Audience for the authentication.")
+
+
+class OIDCCredential(ToolkitResource):
+    read: AuthenticationClientIdSecret | None = Field(
+        default=None, description="Source/Read authentication information."
+    )
+    write: AuthenticationClientIdSecret | None = Field(
+        default=None, description="Destination/Write authentication information."
+    )
+
+
+class TransformationYAML(ToolkitResource):
+    external_id: str = Field(description="The external ID provided by the client.")
+    name: str = Field(description="Name of the transformation.")
+    ignore_null_fields: bool = Field(
+        description="Indicates how null values are handled on updates: ignore or set null."
+    )
+    destination: (
+        DestinationSourceType | DataModelSource | ViewDataSource | RawDataSource | SequenceRowDataSource | None
+    ) = Field(default=None, description="Destination data type.")
+    query: str | None = Field(default=None, description="SQL query of the transformation.")
+    conflict_mode: Literal["abort", "delete", "update", "upsert"] | None = Field(
+        default=None,
+        description="Behavior when the data already exists.",
+    )
+    is_public: bool | None = Field(
+        default=None,
+        description="Indicates if the transformation is visible to all in project or only to the owner.",
+    )
+    authentication: AuthenticationClientIdSecret | OIDCCredential | None = Field(
+        default=None,
+        description="Authentication information for the transformation.",
+    )
+    data_set_external_id: str | None = Field(
+        default=None,
+        description="External ID of the data set to which the transformation belongs.",
+    )
+    tags: list[str] | None = Field(
+        default=None,
+        description="List of tags for the Transformation.",
+        max_length=5,
+    )
+
+    @model_validator(mode="before")
+    def validate_destination_type(cls, values: dict) -> dict:
+        """
+        Validate the destination type and ensure it matches the expected structure.
+        Raises:
+            ValueError: If the destination type is not recognized or if the structure is invalid.
+        """
+        destination = values.get("destination")
+        if destination is None:
+            return values
+
+        dest_type = destination.get("type")
+
+        try:
+            if dest_type == "sequence_rows":
+                SequenceRowDataSource.model_validate(destination)
+
+            elif dest_type == "raw":
+                RawDataSource.model_validate(destination)
+
+            elif dest_type in ["instances"]:
+                DataModelSource.model_validate(destination)
+
+            elif dest_type in ["nodes", "edges"]:
+                ViewDataSource.model_validate(destination)
+
+        except ValidationError as e:
+            raise ValueError(f"Invalid destination data for type '{dest_type}': {e!s}")
```

Comment:
This should return a validated resource. In addition I suggest you use the `__subclasses__()` method with a BaseClass for all the destinations. See how capabilities is implemented.